packages/http-client-java/generator/http-client-generator-test/src/main/java/com/type/model/inheritance/enumdiscriminator/models/Snake.java

```diff
@@ -21,7 +21,7 @@ public class Snake implements JsonSerializable<Snake> {
      * discriminator property
      */
     @Generated
-    private SnakeKind kind = SnakeKind.fromString("Snake");
+    private SnakeKind kind;
```

Comment:
I need to investigate this. Not incorrect, but unexpected.

Reply:
New generated code is correct, previous one is wrong. So this is expected change.

Previous code has [bug](https://github.com/microsoft/typespec/pull/4492#discussion_r1772554245):
The default value is determined by L81: https://github.com/microsoft/typespec/blob/3804d7a5881d86c8e064a41f7eb6b11a070c92eb/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/implementation/PolymorphicDiscriminatorHandler.java#L80-L82

Here the clientType was wrongly mapped to `ChoiceMapper` when mapping client model property:
https://github.com/microsoft/typespec/blob/3804d7a5881d86c8e064a41f7eb6b11a070c92eb/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/SchemaMapper.java#L52-L56

Though we mapped `SnakeKind` correctly in `ClientMapper`: https://github.com/microsoft/typespec/blob/3804d7a5881d86c8e064a41f7eb6b11a070c92eb/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ClientMapper.java#L94-L104

Reply:
But now we seem lot the initialization of this `type` property? It is not in ctor either.

I see, this is a problem of the handling of polymorphic in TypeSpec, that the base class in common case should not be initialized (and does not have a `type` value).

Reply:
Yes. "Snake" is not a valid kind.